docs/how-to/performance_guidelines.rst

```diff
@@ -1,125 +1,103 @@
-concurrently. Both levels of optimization are crucial for achieving maximum
+----------------------
+
+Multiprocessor level optimization involves maximizing parallel execution within
+each multiprocessor on a device. The key to multiprocessor level optimization
+is to efficiently utilize the various functional units within a multiprocessor.
+For example, ensuring a sufficient number of resident warps, so that every clock
+cycle has an instruction from a warp is ready for execution. This instruction could
+either be another independent instruction of the same warp, which exploits
+:ref:`instruction level optimization <instruction-level parallelism>`, or more
+commonly an instruction of another warp, which exploits thread-level parallelism.
+
+On the other hand, device level optimization focuses on the device as a whole,
+aiming at keeping all multiprocessors busy by executing enough kernels
+concurrently. Both multiprocessor and device levels of optimization are crucial for achieving maximum
 performance. They work together to ensure efficient utilization of the
-resources of the GPU, from the individual multiprocessors to the device as a
+GPU resources, ranging from individual multiprocessors to the device as a
 whole.
 
 .. _memory optimization:
-
-Memory optimization
-===================
+Memory throughput optimization
+===============================
 
 The first step in maximizing memory throughput is to minimize low-bandwidth
-data transfers. This involves reducing data transfers between the host and the
-device, as these have lower bandwidth than transfers between global memory and
-the device.
-
-Additionally, data transfers between global memory and the device should be
-minimized by maximizing the use of on-chip memory: shared memory and caches.
-Shared memory acts as a user-managed cache, where the application explicitly
-allocates and accesses it. A common programming pattern is to stage data from
-device memory into shared memory. This involves each thread of a block loading
-data from device memory to shared memory, synchronizing with all other threads
-of the block, processing the data in shared memory, synchronizing again if
-necessary, and writing the results back to device global memory.
+data transfers between the host and the device.
+
+Additionally, maximize the use of on-chip memory, that is, shared memory and caches, and minimize transfers with global memory. Shared memory acts as a user-managed cache explicitly allocated and accessed by the application. A common programming pattern is to stage data from device memory into shared memory. The staging of data from the device to shared memory involves the following steps:
+
+1. Each thread of a block loading data from device memory to shared memory.
+2. Synchronizing with all other threads of the block.
+3. Processing the data stored in shared memory.
+4. Synchronizing again if necessary.
+5. Writing the results back to the device global memory.
```

Comment:
This numbered list doesn't seem to be rendered well. Could you please check this out. It looks like a list with numbers but not a numbered list.

Reply:
For me, this is rendered:

<img width="821" alt="image" src="https://github.com/user-attachments/assets/b383f1a1-e5b7-4eb1-8103-e2a7d2df93b5">

Reply:
Ok